mmv1/products/vertexai/EndpointWithModelGardenDeployment.yaml

```diff
@@ -0,0 +1,972 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: EndpointWithModelGardenDeployment
+api_resource_type_kind: Endpoint
+description: Create an Endpoint and deploy a Model Garden model to it.
+references:
+  guides: null
+  api: 'https://cloud.google.com/vertex-ai/docs/reference/rest/v1beta1/projects.locations/deploy'
+docs: null
+id_format: 'projects/{{project}}/locations/{{location}}/endpoints/{{endpoint}}'
+base_url: 'projects/{{project}}/locations/{{location}}:deploy'
+self_link: 'projects/{{project}}/locations/{{location}}/endpoints/{{endpoint}}'
+create_url: 'projects/{{project}}/locations/{{location}}:deploy'
+import_format:
+- projects/{{project}}/locations/{{location}}/endpoints/{{endpoint}}
+timeouts:
+  insert_minutes: 180
+  delete_minutes: 20
+async:
+  actions: ['create']
+  type: 'OpAsync'
+  operation:
+    timeouts:
+      insert_minutes: 180
+    base_url: '{{op_id}}'
+  result:
+    resource_inside_response: true
+custom_code:
+  post_create: 'templates/terraform/post_create/resource_vertexai_endpoint_with_model_garden_deployment.go.tmpl'
+  custom_delete: 'templates/terraform/custom_delete/resource_vertexai_endpoint_with_model_garden_deployment.go.tmpl'
+examples:
+- name: 'vertex_ai_deploy'
+  primary_resource_id: 'deploy'
+  primary_resource_name: 'fmt.Sprintf("tf_test_example_deploy%s", context["random_suffix"])'
+  vars:
+    project: 'vertex-ai'
+    publisher_model_name: 'publisher_model_name'
+  ignore_read_extra:
+  - 'project'
+  exclude_test: true
```

Comment:
Why exclude test?

Reply:
Unless there's a particular reason the test is invalid we should prefer to test here and in handwritten tests.

This example is used in documentation, so whenever possible we would like to make sure those examples actually work

Reply:
Test is covered by handwritten test and seems like the common practice for other Vertex AI resources.